fastparse/shared/src/test/scala/fastparse/ByteTests.scala

```diff
@@ -267,29 +277,115 @@ object ByteTests extends TestSuite {
+          assert(d.toString == "ByteVector(4 bytes, 0x01020304)")
+
+          // Hex Strings
+          val e = hex"cafebabedeadbeef"
+          assert(e.toString == "ByteVector(8 bytes, 0xcafebabedeadbeef)")
+        }
+        'operations{
+          import fastparse.byte._
+
+          val a = Bytes(0xff, 0xff)
+          val b = Bytes(0x01, 0x01)
+
+          // Simple operations
+          assert(
+            a.length == 2,
+            a(0) == 0xff.toByte,
+            b(0) == 0x01.toByte
+          )
+
+          // Concatenating byte vectors
+          val c = a ++ b
+          assert(c.toString == "ByteVector(4 bytes, 0xffff0101)")
+
+          val d = b ++ a
+          assert(d.toString == "ByteVector(4 bytes, 0x0101ffff)")
+
+          // Converting to Arrays
+          val arr = d.toArray
+          assert(
+            arr(0) == 1, arr(1) == 1,
+            arr(2) == -1, arr(3) == -1
+          )
+
+
+          // Convenient conversions
+          assert(
+            c.toHex == "ffff0101",
+            c.toBase64 == "//8BAQ==",
+            Bytes.fromHex("ffff0101").get == c,
+            Bytes.fromHex("""
+              ff ff
+              01 01
+            """).get == c,
+            Bytes.fromBase64("//8BAQ==").get == c
+          )
+
 
-        assert(BS(0x01, 0xff).toSeq == Array[Byte](0x01.toByte, 0xff.toByte).toSeq)
-        assert(BS("hello":_*).toSeq == Array[Byte](104, 101, 108, 108, 111).toSeq)
+        }
+      }
+      'bytes{
+        'construction{
+          import fastparse.byte._
+
+          // Constructing a short ByteVector from bytes
+          val a = Bytes(0x01, 0xff)
+          assert(a.toString == "ByteVector(2 bytes, 0x01ff)")
+
+          // Constructing a ByteVector from an ASCII string
+          val b  = Bytes("hello":_*)
+          assert(b.toString == "ByteVector(5 bytes, 0x68656c6c6f)")
+
+          // Constructing a ByteVector copying from a Array[Byte]
+          val byteArray = Array[Byte](1, 2, 3, 4)
+          val c = Bytes(byteArray)
+          assert(c.toString == "ByteVector(4 bytes, 0x01020304)")
+
+          // Unsafe construction from an Array[Byte], without copying; assumes
+          // You do not mutate the underlying array, otherwise things break.
+          val d = Bytes.view(byteArray)
+          assert(d.toString == "ByteVector(4 bytes, 0x01020304)")
+        }
       }
       'hexBytes{
         import fastparse.byte._
 
-        assert(hexBytes("01 ff").toSeq == BS(0x01, 0xff).toSeq)
-
-        val multiLine = """
-          de ad be ef
-          ca fe ba be
-        """
-        assert(hexBytes(multiLine).toSeq == BS(
-          0xde, 0xad, 0xbe, 0xef,
-          0xca, 0xfe, 0xba, 0xbe
-        ).toSeq)
+        assert(
+          hex"01 ff" == Bytes.fromHex("01 ff").get,
+          hex"""
+            de ad be ef
+            ca fe ba be
+          """ == Bytes.fromHex("""
+            de ad be ef
+            ca fe ba be
+          """).get
```

Comment:
Maybe it's worth to test `hex""` and `fromHex` independently? Are we sure that they both work correctly?